File: 1448 Count Good Nodes in Binary Tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int goodNodeElement(struct TreeNode* root, int max){
    if (root==NULL) return 0;

    if (root->val > max) max = root->val;

    if (root->val >= max)
        return 1 + goodNodeElement(root->left,max) +
               goodNodeElement(root->right,max);
    return goodNodeElement(root->left,max) +
               goodNodeElement(root->right,max);
}


int goodNodes(struct TreeNode* root){
   if (root==NULL) return 0;
    else return goodNodeElement(root,root->val);
}

File: 1448 Count Good Nodes in Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    
    private int count;
    
    public int goodNodes(TreeNode root) {
        dfs(root, root.val);
        return count;
    }
    
    private void dfs(TreeNode node, int maxSoFar) {
        if (node == null) {
            return;
        }
        if (node.val >= maxSoFar) {
            count++;
        }
        dfs(node.left, Integer.max(maxSoFar, node.val));
        dfs(node.right, Integer.max(maxSoFar, node.val));
    }
}
